restore nRF52 support (#794)
nrfconnect/sdk-sidewalk: doc/releases_and_migration/migration_guide_addon_changelog.rst

Persistent Sidewalk cryptographic keys are now stored in the Key Management Unit (KMU) by default on supported nRF54L Series platforms.
Use the ``CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE_KMU`` Kconfig option to control this feature.

> On nRF52 Series platforms, this option is not available and the default remains settings-based PSA trusted storage backend.

Reviewer: ```suggestion
This option is not available on nRF52 Series platforms, which continue to use the default settings-based PSA trusted storage backend.
```
